### ISSUE TITLE
Change password / generate new parrword

### ISSUE THREAD

**Issue by hemmond** [psztrnk/catalog#5]
Hi, I started using this project (great idea), but I have problem. When installing, I have created my user with my password. After that I wanted to give my girlfriend access to the system (we share our library and we both need access to the list of our books). 

But there is no way I could change password afterwards in the interface. And I cannot modify password even in the database, because it is hashed and salted (which is actually a great thing, thank you for that). 

Adding another user has the same problem, because again - I have no "accessible" way to generate new hash which will be compatible with the system. I know quite a few algorithms how to generate salted password and in the documentation, there are no indications as to which algorithm has been used. 

After studying the code (during writing of the above), I've came up with a dummy script which I ran from terminal on my personal laptop inside clone of this repository. It generated string I could insert into database into database field "password" (maybe it will help to someone with similar problem). 

    <?php
    require('toolkit/bootstrap.php');
    $pwd = "yourRequiredPassword";
    $apppassword = password::hash($pwd);
    echo $apppassword."\n";
    ?>

**Comment by psztrnk** [comment]
Hi hemmond, thanks for posting this. 

Indeed, the app aims to be a single-user application, and there is no way to manage user settings on the UI (Catalog was written to manage my personal library, I just published it for fun and for others to be able to fork it). However in [Catalog2](https://github.com/psztrnk/catalog2) (which I now use at home), there is a way to change password on the UI, and also there's a pretty easy way to reset your password. Unfortunately it is still a singe-user app, but I believe it is more user friendly and other changes can also be implemented more easily. Have a look, if you want.

Also, you are right, Catalog uses the hash function of the Kirby Toolkit, so the script you posted should work perfectly.

**Comment by hemmond** [comment]
The script I left here was already tested and worked in adding second user to the system - which works correctly, so now me and my girlfriend can fill the catalog with data about our books. No other "programming" was required, your system was designed pretty well for single-user application.

**Comment by psztrnk** [comment]
Yes, it was never meant to be a multi-user app, there are no permissions etc. defined.
The app works as a "(semi)multi-user system" only in your use case, when the two of you build the same common collection. Real multi-user would mean to have an ownership, so the visibility of the items would be limited to the owner by default. In this way the two users would be able to build their own collection and eg. decide whether the item is shared with the other users (part of the common collection) or it is only part of their private collection.
